util: don't panic on IPv6 entries in cidr mapping

andrewbaptist opened the pull request:
Previously the code would encounter an index out of bounds if the cidr mapping file had a cidr length greater than 32 bits. This could only happen with IPv6 addresses. Note that if there any invalid mappings the code will display the error in the logs but won't process any of the file.

The code already handled mapping lookups for IPv6, but these code changes also make that more explicit.

Epic: none
Informs: #130814

Release note: None

Commit:
util: don't panic on IPv6 entries in cidr mapping

Previously the code would encounter an index out of bounds if the cidr
mapping file had a cidr length greater than 32 bits. This could only
happen with IPv6 addresses. Note that if there any invalid mappings the
code will display the error in the logs but won't process any of the
file.

The code already handled mapping lookups for IPv6, but these code
changes also make that more explicit.

Epic: none
Informs: #130814

Release note: None

cockroach-teamcity commented:
<!-- Reviewable:start -->
This change is [<img src="https://reviewable.io/review_button.svg" height="34" align="absmiddle" alt="Reviewable"/>](https://reviewable.io/reviews/cockroachdb/cockroach/131221)
<!-- Reviewable:end -->

Commit:
util: remove unused cidr method

We don't use the WrapTLS method and it is better to remove it and if we
need it in the future bring it back.

Epic: none

Release note: None

adwittumuluri reviewed (approved):
thanks for the fix!

andrewbaptist commented:
bors r=adwittumuluri
